pulp_rpm/app/shared_utils.py

```diff
@@ -33,12 +32,26 @@ def _prepare_package(artifact, filename):
         )
 
     package = Package.createrepo_to_dict(cr_pkginfo)
-
     package["location_href"] = filename
+
     artifact_file.close()
     return package
 
 
+def _generate_package_nevra(pkg):
+    """
+    Helper function to generate NEVRA.
+
+    Args:
+        pkg(dict): package metadata
+
+    Returns: NEVRA or NVRA as a string
+    """
+    if int(pkg["epoch"]) > 0:
+        return f"{pkg['name']}-{pkg['epoch']}:{pkg['version']}-{pkg['release']}.{pkg['arch']}.rpm"
+    return f"{pkg['name']}-{pkg['version']}-{pkg['release']}.{pkg['arch']}.rpm"
```

Comment:
What I suggested earlier is to deduplicate this with the implementations in `Package`.  Just make 3 functions, `nevra()`, `nvra()`, and `nevra_short()` (where `nevra_short()` replaces this function and calls into the other two on either side of the if-statement)